components/core/CarouselSidebarData.js

```diff
@@ -458,6 +466,23 @@ class CarouselSidebarData extends React.Component {
     }
   };
 
+  _handleTagDelete = async (tag) => {
+    const response = await Actions.deleteTag({ tag });
+
+    if (response.success) {
+      this.setState({ suggestions: response.tags });
+    }
+
+    if (Events.hasError(response)) {
+      return;
+    }
+  };
+
+  fetchSuggestions = async () => {
```

Comment:
One thing I might suggest for a future iteration of this is: instead of fetching these every time a user opens carousel sidebar data, we can instead fetch them all at once at the time of login (see viewer.js in getById). That way you can access it like this.props.viewer.tags

Or we could fetch it like how you do now (when the first open the editing panel) and then save it to viewer. Then after that if you see that viewer.tags already exists, you don't need to fetch again. That also has a similar effect (let me know if you want more details on how to do that)

Reply:
actually yeah do you think you can make one of those changes before I merge this PR? I noticed that the loading of tags can be a bit slow to do each time 
![image](https://user-images.githubusercontent.com/33686587/111697521-e7756d00-87f2-11eb-9ff5-9ccc01fff614.png)

Reply:
Yeah I noticed that if you are going through the carousel, every time you go to the next item it calls this "get all tags" again, which can be pretty inefficient. I think it only makes sense to call it once you click into the tags input field, if you were to fetch upon first use of edit tags, rather than when the user first logs in

Reply:
Yeah Martina. You're very right. Thank you for pointing this out.